ts/private/ts_project_options_validator.js

```diff
@@ -142,6 +142,15 @@ function main(_a) {
             }
         }
     }
+
+    function check_exclude_and_rootDir() {
+        if (config.exclude === undefined && attrs['root_dir'] !== undefined && attrs['root_dir'] !== '') {
+            console.error('tsconfig validation failed: when root_dir is set, exclude must also be set. See: https://github.com/aspect-build/rules_ts/issues/644 for more details.')
```

Comment:
Should we say "tsconfig exclude"?
Maybe also "when attribute root_dir is set"?

Just to try to distinguish between attributes in BUILD vs keys in tsconfig.json